packages/xdl/src/__tests__/User-test.ts

```diff
@@ -0,0 +1,192 @@
+        'sessionSecret',
+        'session-secret'
+      );
+
+      // Log user out and check if the cache is empty
+      await manager.logoutAsync();
+      expect(await UserSettings.getAsync('auth', null)).toBeNull();
+    });
+  });
+
+  describe('token', () => {
+    afterEach(() => {
+      process.env.EXPO_TOKEN = '';
+    });
+
+    it('authenticates with token', async () => {
+      const api = _newTestApiV2();
+      const manager = _newTestUserManager();
+
+      // Mock authorization token response and fetch the user
+      process.env.EXPO_TOKEN = 'auth-token';
+      api.postAsync.mockResolvedValue({ authorizationToken: 'auth-token' });
+      expect(await manager.getCurrentUserAsync()).toHaveProperty(
+        'authorizationToken',
+        'auth-token'
+      );
+    });
+
+    it(`doesn't cache user when authenticating with credentials`, async () => {
+      const api = _newTestApiV2();
+      const manager = _newTestUserManager();
+      const settingSpy = jest.spyOn(UserSettings, 'setAsync');
+
+      // Mock authorization token response and fetch the user
+      process.env.EXPO_TOKEN = 'auth-token';
+      api.postAsync.mockResolvedValue({ authorizationToken: 'auth-token' });
+      expect(await manager.getCurrentUserAsync()).toHaveProperty(
+        'authorizationToken',
+        'auth-token'
+      );
+      // Note: unfortunately, there is no other way to see if the user data was cached
+      expect(UserSettings.setAsync).not.toBeCalled();
+
+      settingSpy.mockRestore();
+    });
+
+    it('uses token over existing credentials session', async () => {
+      const api = _newTestApiV2();
+      const manager = _newTestUserManager();
+
+      // Mock login response and authenticate
+      api.postAsync.mockResolvedValue({ sessionSecret: 'session-secret' });
+      await manager.loginAsync('user-pass', { username: 'expouser', password: 'expopass' });
+      expect(await manager.getCurrentUserAsync()).toHaveProperty('sessionSecret', 'session-secret');
+
+      // Empty in-memory cache, to force-fetch it
+      manager._currentUser = null;
+
+      // Mock token response and get user
+      process.env.EXPO_TOKEN = 'auth-token';
+      api.postAsync.mockResolvedValue({ authorizationToken: 'auth-token' });
+      const user = await manager.getCurrentUserAsync();
+      expect(user).toHaveProperty('authorizationToken', 'auth-token');
+      expect(user).not.toHaveProperty('sessionSecret', 'session-secret');
+    });
+  });
+});
+
+function _newTestUserManager() {
+  const manager = new UserManagerInstance();
+  manager.initialize();
+  return manager;
+}
+
+function _newTestApiV2() {
+  const api = {
+    sessionSecret: null,
+    authorizationToken: null,
+    getAsync: jest.fn(),
+    postAsync: jest.fn(),
+    putAsync: jest.fn(),
+    patchAsync: jest.fn(),
+    deleteAsync: jest.fn(),
+    uploadFormDataAsync: jest.fn(),
+    _requestAsync: jest.fn(),
+  };
+
+  (ApiV2.clientForUser as jest.Mock).mockReturnValue(api);
```

Comment:
Kind of a dirty way to not-having to add this mock every time we need it. If there is a better way, please do correct me 😄

Reply:
Maybe `ApiV2.clientForUser as jest.MockedFunction<typeof ApiV2.clientForUser>` for more accurate typing, but otherwise seems fine 🤷